packages/react/src/VideoPlayer/VideoPlayer.tsx

```diff
@@ -0,0 +1,194 @@
+  const [trackInformation, setTrackInformation] = useState<TextTrackCueList | undefined>(undefined)
+
+  /* > Play / Pause                                                    */
+  const handleVideoPlayback = () => {
+    if (videoRef.current) {
+      if (videoRef.current.paused) {
+        videoRef.current.play()
+      } else {
+        videoRef.current.pause()
+      }
+    }
+  }
+
+  const handleVideoOnPlaying = () => {
+    videoRef.current?.textTracks[0].cues && setTrackInformation(videoRef.current.textTracks[0].cues)
+  }
+
+  /* > Hide Default Captions                                                   */
+  useEffect(() => {
+    if (videoRef.current) {
+      for (let i = 0; i < videoRef.current.textTracks.length; i++) {
+        videoRef.current.textTracks[0].mode = 'hidden'
+      }
+    }
+  }, [videoRef])
+
+  /* > Set breakpoint classes                                                   */
+
+  useEffect(() => {
+    const handleResize = () => {
+      const breakpoint = videoWrapperRef.current?.getBoundingClientRect().width
+      if (breakpoint && breakpoint < 650) {
+        videoWrapperRef.current.classList.add(styles.VideoPlayer__breakpointSmall)
+      } else {
+        videoWrapperRef.current?.classList.remove(styles.VideoPlayer__breakpointSmall)
+      }
+    }
+
+    const resizeObserver = new ResizeObserver(_ => {
+      if (videoWrapperRef.current) {
+        handleResize()
+      }
+    })
+
+    handleResize()
+
+    const currentRef = videoWrapperRef.current
+
+    resizeObserver.observe(videoWrapperRef.current as Element)
+    return () => {
+      resizeObserver.unobserve(currentRef as Element)
+    }
+  }, [videoWrapperRef, playing])
+
+  return (
+    <div
+      className={clsx(
+        styles.VideoPlayer__container,
+        styles.VideoPlayer__overlays,
+        !playing && styles.VideoPlayer__showOverlays,
+      )}
+      ref={videoWrapperRef}
+    >
+      {/* eslint-disable-next-line jsx-a11y/media-has-caption */}
+      <video
+        ref={videoRef}
+        title={title}
+        controls={false}
+        poster={poster}
+        {...rest}
+        className={clsx(styles.VideoPlayer, className)}
+        onPlay={e => {
+          setPlaying(true)
+          onPlay && onPlay(e)
+        }}
+        onPause={e => {
+          setPlaying(false)
+          onPause && onPause(e)
+        }}
+        onPlaying={e => {
+          handleVideoOnPlaying()
+          onPlaying && onPlaying(e)
+        }}
+        onLoadedMetadata={e => {
+          setTotalTime(videoRef.current?.duration || 0)
+
+          onLoadedMetadata && onLoadedMetadata(e)
+        }}
+        onTimeUpdate={() => setPlayedTime(videoRef.current?.currentTime || 0)}
+      >
+        {children}
+      </video>
+      {title && (
+        <div className={styles.VideoPlayer__title}>
+          {branding && (
+            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" width="48" height="48">
```

Comment:
Is this is the invertocat logo? If it is, could you replace this with the version from Octicons please? We shouldn't need to hardcode these, unless you're applying some customisations to it? Will help reduce footprint of this file too.

Reply:
This is updated in this [SHA](https://github.com/primer/brand/pull/373/commits/c4e331d5b4f9967a0295bc43a82782ec6f61a88f)